### ISSUE TITLE
PR 2158 breaks PHP 7.2.12 on Windows 7 x64

### ISSUE THREAD

**Issue by magikstm** [mholt/caddy#2354]
### 1. What version of Caddy are you using (`caddy -version`)?
0.11.1 (self-compiled on Windows 7 x64 from current master https://github.com/mholt/caddy/commit/ce0988f48a62bad7e2ca9509311ea77545af27b4)

### 2. What are you trying to do?
Use PHP 7.2.12 on Windows 7 x64.

### 3. What is your entire Caddyfile?
```text
:2015 {
	root C:\sites\1
	errors error.log
	log / access_log "{remote} | {user} | {when} | {method} | {uri} | {proto} | {status} | {size} | {latency_ms}"
	#gzip
	on startup C:\php\php-cgi.exe -b 6545 &
	fastcgi / 127.0.0.1:6545 {
		ext .php
	}
}
```

### 4. How did you run Caddy (give the full command and describe the execution environment)?
I double-clicked caddy.exe.

### 5. Please paste any relevant HTTP request(s) here.
I navigated to a file named ```index.php``` containing the text ```index file```. PHP content is irrelevant for this issue. All PHP files break.

### 6. What did you expect to see?
Content of the PHP file.

### 7. What did you see instead (give full error messages and/or log)?
Chrome or Firefox display this:
```Access denied.```

PHP error.log has this error:
```[19-Nov-2018 00:46:05 UTC] PHP Warning:  Unknown: failed to open stream: Permission denied in Unknown on line 0```

### 8. How can someone who is starting from scratch reproduce the bug as minimally as possible?
Run PHP 7.2.12 on Windows 7 x64 with current master.

If I manually revert PR https://github.com/mholt/caddy/pull/2158 PHP works perfectly.

**Comment by mholt** [comment]
Hmm, can you print out the values of PATH_INFO and SCRIPT_FILENAME before and after the PR? And what should it actually be?

**Comment by magikstm** [comment]
Before the PR in a file that echo phpinfo(); I get these :
```
$_SERVER['PATH_INFO'] = /phpinfo.php
$_SERVER['SCRIPT_FILENAME'] = C:\sites\1\phpinfo.php
$_SERVER['SCRIPT_NAME'] = /
```

I can't confirm values after the PR as this page isn't rendered by Caddy. Unfortunately, I don't know these values well and can't confirm what is expected. I would have to investigate further.

I've not tested Caddy 0.11.1 on Linux yet. I can't confirm if it's a Windows 7 x64 issue only.

**Comment by mholt** [comment]
Thanks. Will accept a pull request to fix it!

With tests.

**Comment by magikstm** [comment]
I found how to correct it in my CaddyFile.

I adjusted this:
```
fastcgi / 127.0.0.1:6545 {
	ext .php
}
```

To this:
```
fastcgi / 127.0.0.1:6545 {
	ext .php
	index index.php
	split .php
}
```

The ```split``` value is what corrects it.

Ref: https://caddyserver.com/docs/fastcgi

`split specifies how to split the URL; the split value becomes the end of the first part and anything in the URL after it becomes part of the PATH_INFO CGI variable.`

I'm closing this. Sorry for reporting it as an issue.
